crowd_anki/github/github_importer.py

```diff
@@ -32,24 +28,15 @@ def import_from_github(self):
         repo, ok = QInputDialog.getText(None, 'Enter GitHub repository',
                                         'Path:', text='<name>/<repository>')
         if repo and ok:
-            self.download_and_import(repo)
+            self.download_and_import_git(repo)
 
-    def download_and_import(self, repo):
+    def download_and_import_git(self, github_repo):
+        repo_parts = github_repo.split("/")
+        repo_dir = Path(self.collection.media.dir()).joinpath("..", "CrowdAnkiGit", repo_parts[-1], "git")
         try:
-            response = urlopen(GITHUB_LINK.format(repo))
-            response_sio = BytesIO(response.read())
-            with zipfile.ZipFile(response_sio) as repo_zip:
-                repo_zip.extractall(tempfile.tempdir)
-
-            deck_base_name = repo.split("/")[-1]
-            deck_directory_wb = Path(tempfile.tempdir).joinpath(deck_base_name + "-" + BRANCH_NAME)
-            deck_directory = Path(tempfile.tempdir).joinpath(deck_base_name)
-            utils.fs_remove(deck_directory)
-            deck_directory_wb.rename(deck_directory)
-            # Todo progressbar on download
-
-            AnkiJsonImporter.import_deck_from_path(self.collection, deck_directory)
-
-        except (URLError, HTTPError, OSError) as error:
-            aqt.utils.showWarning("Error while trying to get deck from GitHub: {}".format(error))
-            raise
+            repo_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
why do we do mkdir in the pull branch? presumably it should be in the clone branch?